src/test/java/org/apache/commons/collections4/MapBuilderTest.java

```diff
@@ -0,0 +1,122 @@
+package org.apache.commons.collections4;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Test Cases for Map Builder
+ */
+class MapBuilderTest {
```

Comment:
IMO, you do not/should not need to test the behavior of Map implementations. Checking the type of the built instance is enough.

Reply:
I drifted towards functional testing here.Just to clarify checking instance type as MapType expected (LinkedHashMap/TreeMap) etc should be done here ?